libr/anal/types.c

```diff
@@ -40,7 +40,105 @@ R_API void r_anal_type_del(RAnal *anal, const char *name) {
 	sdb_unset (DB, str, 0);
 }
 
-R_API char* r_anal_type_to_str(RAnal *a, const char *type) {
+R_API int r_anal_type_get_size (RAnal *anal, char *type) {
+	const char *t = sdb_const_get (anal->sdb_types, type, 0);
+	if (!strcmp (t, "type")){
+		char *query = r_str_newf ("type.%s.size", type);
+		int ret;
+		ret = r_num_math (NULL, sdb_const_get (anal->sdb_types, query, 0));
```

Comment:
maybe r_num_get is enough? do this expression contain operations or it is just a tnumber? if its just a number you can use the sdb_num api and avoid extra steps